homeassistant/components/lcn/sensor.py

```diff
@@ -1,63 +1,76 @@
     VARIABLES,
 )
-from .helpers import get_connection
+from .helpers import get_device_connection
+
 
+def create_lcn_sensor_entity(hass, entity_config, config_entry):
+    """Set up an entity for this domain."""
+    host_id = config_entry.entry_id
+    device_connection = get_device_connection(
+        hass, entity_config[CONF_UNIQUE_DEVICE_ID], config_entry
+    )
 
-async def async_setup_platform(
-    hass, hass_config, async_add_entities, discovery_info=None
-):
-    """Set up the LCN sensor platform."""
-    if discovery_info is None:
-        return
+    if (
+        entity_config[CONF_DOMAIN_DATA][CONF_SOURCE]
+        in VARIABLES + SETPOINTS + THRESHOLDS + S0_INPUTS
+    ):
+        return LcnVariableSensor(entity_config, host_id, device_connection)
+    # in LED_PORTS + LOGICOP_PORTS
+    return LcnLedLogicSensor(entity_config, host_id, device_connection)
 
-    devices = []
-    for config in discovery_info:
-        address, connection_id = config[CONF_ADDRESS]
-        addr = pypck.lcn_addr.LcnAddr(*address)
-        connections = hass.data[DATA_LCN][CONF_CONNECTIONS]
-        connection = get_connection(connections, connection_id)
-        device_connection = connection.get_address_conn(addr)
 
-        if config[CONF_SOURCE] in VARIABLES + SETPOINTS + THRESHOLDS + S0_INPUTS:
-            device = LcnVariableSensor(config, device_connection)
-        else:  # in LED_PORTS + LOGICOP_PORTS
-            device = LcnLedLogicSensor(config, device_connection)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up LCN switch entities from a config entry."""
+    entities = []
 
-        devices.append(device)
+    for entity_config in config_entry.data[CONF_ENTITIES]:
+        if entity_config[CONF_DOMAIN] == DOMAIN_SENSOR:
+            entities.append(create_lcn_sensor_entity(hass, entity_config, config_entry))
 
-    async_add_entities(devices)
+    async_add_entities(entities)
 
 
 class LcnVariableSensor(LcnEntity):
     """Representation of a LCN sensor for variables."""
 
-    def __init__(self, config, device_connection):
+    def __init__(self, config, host_id, device_connection):
         """Initialize the LCN sensor."""
-        super().__init__(config, device_connection)
+        super().__init__(config, host_id, device_connection)
 
-        self.variable = pypck.lcn_defs.Var[config[CONF_SOURCE]]
-        self.unit = pypck.lcn_defs.VarUnit.parse(config[CONF_UNIT_OF_MEASUREMENT])
+        self.variable = pypck.lcn_defs.Var[config[CONF_DOMAIN_DATA][CONF_SOURCE]]
+        self.unit = pypck.lcn_defs.VarUnit.parse(
+            config[CONF_DOMAIN_DATA][CONF_UNIT_OF_MEASUREMENT]
+        )
 
         self._value = None
 
     async def async_added_to_hass(self):
         """Run when entity about to be added to hass."""
         await super().async_added_to_hass()
-        await self.device_connection.activate_status_request_handler(self.variable)
+        if not self.device_connection.is_group:
```

Comment:
Is this a separate change or something related to the move to config entries? Just looking for a hint to make review faster before I dig in.

Reply:
It is a necessary change for the next version of the underlying library. The `activate_status_request_handler` method will be removed for groups as it makes no sense. The version currently in use is not bothered by this.